addons/web/static/src/js/views/calendar/calendar_renderer.js

```diff
@@ -881,7 +881,10 @@ return AbstractRenderer.extend({
         for (const event of events) {
             const start = moment(event.extendedProps.r_start);
             const end = moment(event.extendedProps.r_end);
-            const key = this._getFormattedDate(start, end, false, event.extendedProps.record.allday);
+            // get the original event to get the real allDay value as
+            // the fullcalendar year view enforce allDay to true
+            const originalEvent = this.state.data.find(e => e.id === event.extendedProps.record.id);
+            const key = this._getFormattedDate(start, end, false, originalEvent.allDay);
```

Comment:
@mcm-odoo or @jpp-odoo 
Now really ready to review :-)

These lines are sadly necessary because the fullcalendar year view plugin overrides the allDay value to true. I suppose it is like that for other reasons i.e. the create context. I did not investigate this matter, as this is a fix in a stable version and the view is completely rewritten started with 16.0 branch.

Reply:
I could also make a map beforehand, to avoid to loop on state data for all events of a specific date.